thundra/listeners/aws_xray_listeners.py

```diff
@@ -1,89 +1,104 @@
         if dict_key in dictionary:
             value = dictionary[dict_key]
             if isinstance(value, str) or isinstance(value, int) or isinstance(value, bool):
-                document.put_annotation(self.normalize_annotation_name(key), self.normalize_annotation_value(value))
+                document.put_annotation(self._normalize_annotation_name(key), self._normalize_annotation_value(value))
 
-    def add_annotation(self, span):
-        self.xray_data = AWSXrayListener._data
+    def _add_annotation(self, span):
+        self.xray_data = AWSXRayListener._data
         document = xray_recorder.current_subsegment()
         if document:
             document.put_annotation('traceId', span.context.trace_id)
             document.put_annotation('transactionId', span.context.transaction_id)
             document.put_annotation('spanId', span.context.span_id)
             document.put_annotation('parentSpanId', span.context.parent_span_id)
             span_dict = vars(span)
-            self.put_annotation_if_available('domainName', span_dict, 'domain_name', document)
-            self.put_annotation_if_available('className', span_dict, 'class_name', document)
-            self.put_annotation_if_available('operationName', span_dict, 'operation_name', document)
-            self.put_annotation_if_available('startTimeStamp', span_dict, 'start_time', document)
-            self.put_annotation_if_available('finishTimeStamp', span_dict, 'finish_time', document)
+            self._put_annotation_if_available('domainName', span_dict, 'domain_name', document)
+            self._put_annotation_if_available('className', span_dict, 'class_name', document)
+            self._put_annotation_if_available('operationName', span_dict, 'operation_name', document)
+            self._put_annotation_if_available('startTimeStamp', span_dict, 'start_time', document)
+            self._put_annotation_if_available('finishTimeStamp', span_dict, 'finish_time', document)
             duration = int(span.get_duration())
             document.put_annotation('duration', duration)
 
             if self.xray_data is not None:
                 for key in self.xray_data:
-                    self.put_annotation_if_available(key, self.xray_data, key, document)
+                    self._put_annotation_if_available(key, self.xray_data, key, document)
 
-    def add_metadata(self, span):
+    def _add_metadata(self, span):
         document = xray_recorder.current_subsegment()
         if document:
             for key in span.tags:
                 document.put_metadata(key, span.tags[key])
 
-    def normalize_operation_name(self, operation_name):
+    def _normalize_operation_name(self, operation_name):
         if operation_name:
             return operation_name[0:200]
         return constants.AwsXrayConstants['DEFAULT_OPERATION_NAME']
 
-    def normalize_annotation_value(self, value):
+    def _normalize_annotation_value(self, value):
```

Comment:
Can you also fix this logic according to allowed characters in the AWS X-Ray doc?